make-release.sh

```diff
@@ -11,147 +11,221 @@
-    git push origin "${PR_BRANCH}"
+    git pull origin "${PR_BRANCH}" || true
+    $DRY_RUN git push origin "${PR_BRANCH}"
     lastCommitComment="$(git log -1 --pretty=%B)"
-    hub pull-request -f -m "${lastCommitComment}" -b "${BUMP_BRANCH}" -h "${PR_BRANCH}"
+    $DRY_RUN hub pull-request -f -m "${lastCommitComment}" -b "${BUMP_BRANCH}" -h "${PR_BRANCH}"
   fi
-  set -e
   git checkout "${CURRENT_BRANCH}"
 }
 
-usage ()
-{
-  echo "Usage: $0 --version [VERSION TO RELEASE]"
-  echo "Example: $0 --version v0.1.0"; echo
-}
+prerelease() {
+  echo "[INFO] Starting prerelease procedure"
+  # derive bugfix branch from version
+  BRANCH=${VERSION#v}
+  BRANCH=${BRANCH%.*}.x
 
-if [[ ! ${VERSION} ]]; then
-  usage
-  exit 1
-fi
+  echo "[INFO] Creating ${BRANCH} from ${MAIN_BRANCH}"
+  git checkout ${MAIN_BRANCH}
+  git branch "${BRANCH}" || git checkout "${BRANCH}"
+  sed -i version/version.go -r -e "s#(Version = \")(v?[0-9.]+-rc)(\")#\1${VERSION}\3#g"
+  if [[ ! -z $(git status -s) ]]; then # dirty
+    COMMIT_MSG="[prerelease] Remove rc from version"
+    git commit -asm "${COMMIT_MSG}"
+  fi
+  $DRY_RUN git push origin "${BRANCH}"
+  git fetch origin "${BRANCH}:${BRANCH}" || true
+  git checkout "${BRANCH}"
 
+  # change VERSION file + commit change into ${BASEBRANCH} branch
+  # bump the y digit, if it is a major release
+  [[ $BRANCH =~ ^([0-9]+)\.([0-9]+)\.x ]] && BASE=${BASH_REMATCH[1]}; NEXT=${BASH_REMATCH[2]}; (( NEXT=NEXT+1 )) # for BRANCH=0.1.x, get BASE=0, NEXT=2
+  NEXT_VERSION_Y="${BASE}.${NEXT}.0"
+  bump_version "${NEXT_VERSION_Y}" "${MAIN_BRANCH}"
+  echo "Prerelease is done"
+}
 
-# derive bugfix branch from version
-BRANCH=${VERSION#v}
-BRANCH=${BRANCH%.*}.x
+release() {
+  echo "[INFO] Starting Release procedure"
+  # derive bugfix branch from version
+  BRANCH=${VERSION#v}
+  BRANCH=${BRANCH%.*}.x
 
-# if doing a .0 release, use main branch; if doing a .z release, use $BRANCH
-if [[ ${VERSION} == *".0" ]]; then
-  BASEBRANCH="${MAIN_BRANCH}"
-else
   BASEBRANCH="${BRANCH}"
-fi
 
-# work in tmp dir
-if [[ $TMP ]] && [[ -d $TMP ]]; then
-  pushd "$TMP" > /dev/null || exit 1
+  git remote show origin
+
   # get sources from ${BASEBRANCH} branch
-  echo "Check out ${REPO} to ${TMP}/${REPO##*/}"
-  git clone "${REPO}" -q
-  cd "${REPO##*/}" || exit 1
-fi
+  git fetch origin "${BASEBRANCH}":"${BASEBRANCH}" || true
+  git checkout "${BASEBRANCH}"
+
+  # create new branch off ${BASEBRANCH} (or check out latest commits if branch already exists), then push to origin
+  if [[ "${BASEBRANCH}" != "${BRANCH}" ]]; then
+    git branch "${BRANCH}" || git checkout "${BRANCH}"
+    $DRY_RUN git push origin "${BRANCH}"
+    git fetch origin "${BRANCH}:${BRANCH}" || true
+    git checkout "${BRANCH}"
+  else
+    git fetch origin "${BRANCH}:${BRANCH}" || true
+    git checkout "${BRANCH}"
+  fi
 
-git remote show origin
+  # change version/version.go file
+  sed -i version/version.go -r -e "s#(Version = \")(v?[0-9.]+)(\")#\1${VERSION}\3#g"
```

Comment:
This should be doable without grouping:
```suggestion
  sed -i version/version.go -e "s#Version = \".*\"#Version = \"${VERSION}\"#g"
```
The current regex will also fail to match the version if it is formatted e.g. `v0.1.0+dev` since it only looks for numbers and dots.

We could also avoid a bit of repetition by doing a line match + replace:
```
sed version/version.go -e "/Version =/ s|\".*\"|\"${VERSION}\"|"
```

Reply:
removed groups for sed in version.go